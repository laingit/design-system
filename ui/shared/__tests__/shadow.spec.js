// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

/* eslint-env jest */
import React from 'react';
import Shadow from '../shadow/';
import renderer from 'react-test-renderer';

describe('ShadowRoot', () => {
  it('creates a custom element', () => {
    const wrapper = renderer
      .create(
        <Shadow.on name="test">
          <div>content</div>
        </Shadow.on>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('custom element has a shadow boundary', () => {
    const wrapper = renderer
      .create(
        <Shadow.on name="test">
          <div>content</div>
        </Shadow.on>
      )
      .toTree();
    expect(wrapper.props.children).toBe(null);
  });
  it('renders with no shadow boundary via config', () => {
    const wrapper = renderer
      .create(
        <Shadow.off name="test">
          <div>content</div>
        </Shadow.off>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with no shadow boundary via prop', () => {
    const wrapper = renderer
      .create(
        <Shadow.on name="test" shadow={false}>
          <div>content</div>
        </Shadow.on>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
