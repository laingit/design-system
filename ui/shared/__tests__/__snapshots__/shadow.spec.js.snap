// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowRoot creates a custom element 1`] = `<lightning-test />`;

exports[`ShadowRoot renders with no shadow boundary via config 1`] = `
<div>
  content
</div>
`;

exports[`ShadowRoot renders with no shadow boundary via prop 1`] = `
<div>
  content
</div>
`;
